l10n_br_account_product/models/account.py

```diff
@@ -198,6 +201,76 @@ def compute_all(self, cr, uid, taxes, price_unit, quantity,
+                        if tx['domain'] == 'icmsfcp']
+        result_fcp = self._compute_tax(cr, uid, specific_fcp, total_base,
+                                       product, quantity, precision, base_tax)
+        totaldc += result_fcp['tax_discount']
+        calculed_taxes += result_fcp['taxes']
+
+        # Calcula ICMS Interestadual (DIFAL)
+        specific_icms_inter = [tx for tx in result['taxes']
+                               if tx['domain'] == 'icmsinter']
+        result_icms_inter = self._compute_tax(
+            cr,
+            uid,
+            specific_icms_inter,
+            total_base,
+            product,
+            quantity,
+            precision,
+            base_tax)
+
+        if (specific_icms_inter and fiscal_position and
+                partner.partner_fiscal_type_id.ind_ie_dest == '9'):
+            if fiscal_position.cfop_id.id_dest == '2':
+                try:
+
+                    # Calcula o DIFAL total
+                    result_icms_inter['taxes'][0]['amount'] = round(
+                        specific_icms_inter[0]['amount'] -
+                        icms_value, precision)
+
+                    # Cria uma chave com o ICMS de intraestadual
+                    result_icms_inter['taxes'][0]['icms_origin_percent'] = \
+                        specific_icms[0]['percent']
+
+                    # Procura o percentual de partilha vigente
+                    icms_partition_ids = self.pool.get(
+                        'l10n_br_tax.icms_partition').search(
+                            cr, uid,
+                            [('date_start', '<=',
+                              time.strftime(DEFAULT_SERVER_DATE_FORMAT)),
+                             ('date_end', '>=',
+                              time.strftime(DEFAULT_SERVER_DATE_FORMAT))])
+
+                    # Calcula o difal de origin e destino
+                    if icms_partition_ids:
+                        icms_partition = self.pool.get(
+                            'l10n_br_tax.icms_partition').browse(
+                                cr, uid, icms_partition_ids[0])
+                        result_icms_inter['taxes'][0]['icms_part_percent'] = \
+                            icms_partition.rate / 100
+
+                        result_icms_inter['taxes'][0]['icms_dest_value'] = \
+                            round(
+                                result_icms_inter['taxes'][0]['amount'] *
+                                (icms_partition.rate / 100),
+                                precision)
+                        result_icms_inter['taxes'][0]['icms_origin_value'] = \
+                            round(
+                                result_icms_inter['taxes'][0]['amount'] *
+                                ((100 - icms_partition.rate) / 100),
+                                precision)
+
+                    # Atualiza o imposto icmsinter
+                    calculed_taxes += result_icms_inter['taxes']
+
+                except:
```

Comment:
kind of strange to catch for any Exception here. Can we not catch more specific exceptions or test some specific assertions? Supposing such a UserError is raised it's not going to be easy to find out what is wrongly set...